### PR TITLE
[4.0] Refactored admin-item-edit_container.es6.js to no longer use jQuery anymore

### PR DESCRIPTION
**Pull request by sven-ahrens**
### Summary of Changes
Removed any jQuery code and rebuilt it with ECMAScript 6.

### Testing Instructions

1. Login into the backend of your joomla! site

2. (Toggle the side navigation if hidden)

3. Click on "Menus" and choose an administrator menu if available. If not, you'll need to create one

4. After being redirected to the "Menus: Items" page: Click on "new" to create an new item

5. Now you should be able to select the menu item type. Click on "select"

6. Choose: "System Links". A dropdown will open. There, you have to choose: "Components Menu Container"

7. At the middle of the page should be a list in form of a tree.

### Expected result
The list has sevaral "root" elements, which is, a list-item, which contains a subtree. 
Current "root elements" on this page are: "All", "Banners", "Contacts", "Messaging" and "News Feeds".
Clicking on one of them should instantly check and disable all elements, which are hold inside of this root.

For example: Clicking on "All" should check and disable every checkbox on the screen. Of course, if you click it again, every checkbox will be unchecked and enabled. 

* Check means, that the checkbox color and text will change to red
* disable means, that the checkbox is still red, but lighter due to less opacity

If a checkbox is either checked nor disabled it's default color is green.

Clicking on a item, which is not a root, will be checked(color switches to red) but not disabled, because you wouldn't be able to click it again. 

Now apply the patch  from this PR , re-run npm and verify that the functionality described above still works as expected.

### CODE REVIEW

File: build/media_src/com_menus/js/admin-item-edit_container.es6.js [hunk 1]
```diff
@@ -2,22 +2,53 @@
...
-
-  const propagate = () => {
-    const $this = $(this);
-    const sub = $this.closest('li').find('.treeselect-sub [type="checkbox"]');
-    sub.prop('checked', this.checked);
-    if ($this.val() === 1) {
-      sub.each(propagate);
-    } else {
-      sub.attr('disabled', this.checked ? 'disabled' : null);
+/**
```
Comment (laoneo) [inline build/media_src/com_menus/js/admin-item-edit_container.es6.js]: Two times copyright

**Comment by SniperSister** [comment]
I have tested this item :white_check_mark: successfully on 252e73053cc534c70bb18bc5edb803b58c70b9a3<br /><br />Works as described! Note for other testers: the expected behavior is broken in the current j4 branch, for reference look at the behavior in a 3.x installation<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22667">issues.joomla.org/tracker/joomla-cms/22667</a>.</sub>

**Comment by dgrammatiko** [comment]
Array.from needs to be polyfilled. Use [].slice.call() instead which works out of the box even in older browsers eg ie

**Comment by laoneo** [comment]
Very good work! Thanks.

**Pull request merged**